File: frontend/src/types/history.ts
import type { AssigneeSummary, IncidentSeverity, IncidentStatus, ProjectSummary } from "./incident";

export interface HistoryIncidentItem {
  id: string;
  incident_number: string;
  title: string;
  project: ProjectSummary;
  severity: IncidentSeverity;
  status: IncidentStatus;
  environment: string | null;
  predicted_category: string | null;
  ai_confidence: number | null;
  detected_at: string;
  resolved_at: string | null;
  resolution_summary: string | null;
  root_cause_summary: string | null;
  current_assignee: AssigneeSummary | null;
}

File: backend/app/schemas/history.py
"""Incident history API schemas."""

from __future__ import annotations

from datetime import datetime
from uuid import UUID

from pydantic import BaseModel

from app.schemas.incident import AssigneeSummary, ProjectSummary


class HistoryIncidentItem(BaseModel):
    id: UUID
    incident_number: str
    title: str
    project: ProjectSummary
    severity: str
    status: str
    environment: str | None = None
    predicted_category: str | None = None
    ai_confidence: float | None = None
    detected_at: datetime
    resolved_at: datetime | None = None
    resolution_summary: str | None = None
    root_cause_summary: str | None = None
    current_assignee: AssigneeSummary | None = None
